packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -76,18 +80,29 @@ + (BOOL)requiresMainQueueSetup {
     return YES;
 }
 
-RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(horizontalSizeClass)
-{
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(horizontalSizeClass:(id)reactTag) {
+    RCTUnsafeExecuteOnMainQueueSync(^{
+        if ([reactTag isKindOfClass:[NSNumber class]]) {
+            UIView *view = [[[self bridge] uiManager] viewForReactTag:reactTag];
+            NSString *horizontalSizeClass = RCTHorizontalSizeClassPreference([view traitCollection]);
+            self->_horizontalSizeClass = horizontalSizeClass;
+        }
+    });
     return _horizontalSizeClass;
 }
 
-RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(userInterfaceLevel)
-{
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(userInterfaceLevel:(id)reactTag) {
+    RCTUnsafeExecuteOnMainQueueSync(^{
+        if ([reactTag isKindOfClass:[NSNumber class]]) {
+            UIView *view = [[[self bridge] uiManager] viewForReactTag:reactTag];
+            NSString *horizontalSizeClass = RCTHorizontalSizeClassPreference([view traitCollection]);
+            self->_horizontalSizeClass = horizontalSizeClass;
```

Comment:
should be `userInterfaceLevel`, not `horizontalSizeClass`?

Reply:
Oooh yes thank you